code/game/objects/structures/undetonated_nuke.dm

```diff
@@ -0,0 +1,59 @@
+/obj/structure/undetonated_nuke
+	name = "undetonated nuclear bomb"
+	desc = "This rusted bomb looks like it's been here for decades. Flaking paint on the side depicts the battle standard of the Traditionalist Coalition. Something is written in Sinta'Azaziba on the side."
+	icon = 'icons/obj/unathi_ruins.dmi'
+	icon_state = "bomb"
+	density = TRUE
+	anchored = FALSE
+	var/can_explode = TRUE
+
+/obj/structure/undetonated_nuke/get_examine_text(mob/user, distance)
+	. = ..()
+	if(GLOB.all_languages[LANGUAGE_AZAZIBA] in user.languages)
+		. += SPAN_NOTICE("The inscription reads \"WARNING: FISSILE MATERIAL. HANDLE WITH CARE.\" Underneath are a few words, scratched into the metal. They read \"IF FOUND, RETURN TO SKALAMAR AT HIGH VELOCITY\"")
+
+/obj/structure/undetonated_nuke/ex_act(severity)
+	if(!can_explode)
+		return
+	switch(severity)
+		if(1.0)
+			explode()
+			return
+		if(2.0)
+			if(prob(50))
+				explode()
+				return
+		if(3.0)
+			return
```

Comment:
what if severity is 1.5 or 2.5 or any other number that is not exactly 1.0 or 2.0 or 3.0

Reply:
it won't explode then

Reply:
Severity for ex_act is rounded to a whole number from 1-3 IIRC, and this is pretty much identical to how most other ex_acts in the game are handled

Reply:
fair